### PR TITLE
feat: added cart management

### PR DESCRIPTION
**Pull request by eakansh125**
I have created cart management but not added:
1. Quantity increment decrement - Will resolve the issues im facing and add that later
2. Move to wishlist - Will add this along with wish list management

Files to look for:
1. Context\cart-context.js
2. Context\cart-reducer.js
3. Components\Cart\Cart.js
4. Components\Cart\CartPrice.js
5. Components\Cart\CartProduct.js
6. Components\Cart\Cart.css
7. Components\Product\ProductCard.js
8. App.js - Route for cart
9. index.js - Added Context

Please review it and approve soon if possible :)

### CODE REVIEW

File: src/Components/Cart/CartProduct.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                                            <button className="btn-outlined margin-lr">-</button>
+                                            <p>1</p>
+                                            <button 
+                                            // onClick={() => {
+                                            //     dispatch({
+                                            //         type: "INC_PRODUCT",
+                                            //         payload: product
+                                            //     })
+                                            // }} 
+                                            className="btn-outlined margin-lr">+</button>
```
Comment (omkarpatke) [inline src/Components/Cart/CartProduct.js:28]: suggestion - remove commented code when adding github.

File: src/Components/Product/ProductCard.js [hunk 2]
```diff
@@ -1,6 +1,8 @@
+import { useCart } from '../../Context/cart-context';
 import './ProductListing.css';
 
-function ProductCard({title, price}) {
+function ProductCard({prod}) {
```
Comment (omkarpatke) [inline src/Components/Product/ProductCard.js:4]: suggestion - use better naming here. so it looks nice.

**Review by omkarpatke — APPROVED** [review]
code looks good, clean, and readable only remove commented code and do proper variable naming.

File: src/Components/Cart/CartProduct.js (hunk 1 above)
Comment (eakansh125) [reply src/Components/Cart/CartProduct.js:28]: sure, have kept it for future references, will uncomment once required. Thanks!

File: src/Components/Cart/CartProduct.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    const { dispatch} = useCart();
+    return (
+         <div className="cart-item flex-row margin-tb">
+                            <div className="cart-item-image-container">
+                                <img className="cart-item-image" src="/img/pexels-ron-lach-10397680.jpg" alt="shirt" />
+                            </div>
+                            <div className="cart-item-desc-container">
+                                <div className="cart-item-desc">
+                                    <h3>{product.title}</h3>
+                                    <h3>${product.price}</h3>
```
Comment (Naman-Saxena1) [inline src/Components/Cart/CartProduct.js:14]: Can destructure product at top of file, won't have to prefix product. before everything

File: src/Context/cart-context.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  const [state, dispatch] = useReducer(cartReducer, {
+    products: products,
+    cart: []
+  })
+  return <CartContext.Provider value={{state, dispatch}}>{children}</CartContext.Provider>
+}
+
+export const useCart = () => useContext(CartContext)
+
+export default CartProvider
```
Comment (Naman-Saxena1) [inline src/Context/cart-context.js]: I think CartContextProvider would be better name but CartProvider also looks fine

**Review by Naman-Saxena1 — APPROVED** [review]
Overall code quality looks good.
It would be great if you could provide preview link to test functionality or at least screenshots in PR description 👍

File: src/Components/Cart/CartProduct.js (hunk 3 above)
Comment (eakansh125) [reply src/Components/Cart/CartProduct.js:14]: Actually the product object is required to be sent through dispatch which is why didn't destructure it

File: src/Context/cart-context.js (hunk 4 above)
Comment (eakansh125) [reply src/Context/cart-context.js]: sure will change that

**Pull request merged**